<a href="https://colab.research.google.com/github/sumaiyasiddiqa/Play-store-app-review-analysis/blob/main/Play_Store_App_Review_Analysis_(Sumaiya_Siddiqa).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

Google Play Store or formerly Android Market, is a digital distribution service developed and operated by Google. 
It is an official apps store that provides variety content such as apps, books, magazines, music, movies and television programs.
It serves an as platform to allow users with 'Google certified' Android operating system devices to donwload applications developed 
and published on the platform either with a charge or free of cost. With the rapidly growth of Android devices and apps,
it would be interesting to perform data analysis on the data to obtain valuable insights.

Lets look at the data which consists of two files:


**1.   playstoredata.csv**:It contains all the detials of the apps on play store

**2.   userreview.csv**  :It contains user reviews for each app.The text in each review has been pre-processed and attributed with three new features: Sentiment (Positive, Negative or Neutral), Sentiment Polarity and Sentiment Subjectivity. 








# Exploring Play Store Data





In [2]:
#import libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
from datetime import datetime
import plotly 
plotly.offline.init_notebook_mode(connected=True)
import plotly.graph_objs as go
import warnings
warnings.filterwarnings("ignore")


In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
#Load th csv file
file_path="/content/drive/MyDrive/CAPSTONE1(play store app review analysis)/Play Store Data.csv"
ps_df=pd.read_csv(file_path)

In [5]:
#Lets randomly look at the rows of the dataset
ps_df[10:20]

App        Category  Rating  \
10                          Text on Photo - Fonteee  ART_AND_DESIGN     4.4   
11          Name Art Photo Editor - Focus n Filters  ART_AND_DESIGN     4.4   
12                   Tattoo Name On My Photo Editor  ART_AND_DESIGN     4.2   
13                            Mandala Coloring Book  ART_AND_DESIGN     4.6   
14  3D Color Pixel by Number - Sandbox Art Coloring  ART_AND_DESIGN     4.4   
15                  Learn To Draw Kawaii Characters  ART_AND_DESIGN     3.2   
16     Photo Designer - Write your name with shapes  ART_AND_DESIGN     4.7   
17                         350 Diy Room Decor Ideas  ART_AND_DESIGN     4.5   
18                    FlipaClip - Cartoon animation  ART_AND_DESIGN     4.3   
19                                     ibis Paint X  ART_AND_DESIGN     4.6   

   Reviews  Size     Installs  Type Price Content Rating        Genres  \
10   13880   28M   1,000,000+  Free     0       Everyone  Art & Design   
11    8788   12M   1,000,000+  Free     0       Everyone  Art & Design   
12   44829   20M  10,000,000+  Free     0           Teen  Art & Design   
13    4326   21M     100,000+  Free     0       Everyone  Art & Design   
14    1518   37M     100,000+  Free     0       Everyone  Art & Design   
15      55  2.7M       5,000+  Free     0       Everyone  Art & Design   
16    3632  5.5M     500,000+  Free     0       Everyone  Art & Design   
17      27   17M      10,000+  Free     0       Everyone  Art & Design   
18  194216   39M   5,000,000+  Free     0       Everyone  Art & Design   
19  224399   31M  10,000,000+  Free     0       Everyone  Art & Design   

        Last Updated Current Ver   Android Ver  
10  October 27, 2017       1.0.4    4.1 and up  
11     July 31, 2018      1.0.15    4.0 and up  
12     April 2, 2018         3.8    4.1 and up  
13     June 26, 2018       1.0.4    4.4 and up  
14    August 3, 2018       1.2.3    2.3 and up  
15      June 6, 2018         NaN    4.2 and up  
16     July 31, 2018         3.1    4.1 and up  
17  November 7, 2017         1.0    2.3 and up  
18    August 3, 2018       2.2.5  4.0.3 and up  
19     July 30, 2018       5.5.4    4.1 and up

In [6]:
ps_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [7]:
#Finding the names of the columns and the number of rows and columns
print(list(ps_df.columns))
print(f'The number of rows of the dataframe playstore data are :{ps_df.shape[0]}')
print(f'The number of columns of the dataframe playstore data are: {ps_df.shape[1]}')

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
The number of rows of the dataframe playstore data are :10841
The number of columns of the dataframe playstore data are: 13


The description of columns of playstore dataframe :
   



> 

1.    App : It has the name of the application.
2.    Category : It has the category to which the app belongs.
3.    Rating :  It has the average rating given by the users in the play store.
4.    Reviews : It tells us about the total number of users who have given a review for the application.
5.    Size : It has the size of the app.
6.    Installs : It has the total no of downloads or installs.
7.    Type : It tells about the type of the app whether free or paid.
8.    Price : It has the price of the app.
9.    Content rating : It states whether or not an app is suitable for all age groups or not. 
10.   Genres : The genre of the app
11.   Last updated : It has the date on which the app was last updated.
12.   Current ver : It has the current version of the app.
13.   Android ver : It tells about the minimum android version required to run the app.










## **Data cleaning**

Removing the null values and duplicate values present in the dataset.By looking at the dataset carefully we can see that their are many special characters and letters in the columns like (Install, Price,Size) which make their conversion to numeric datatype difficult.Let's clean the data by removing all these.



In [8]:
#Check on the null values
ps_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

By above analysis we can see that 'Ratings' column has 1474 null values which is 13% of the data of ratings column, column 'Type' has 1 null value, column 'Content rating' has 1 null value, column 'Current ver' has 8 null values and column 'Android ver' has 3 null values.


Lets first deal with the columns which contains lesser number of NaN values. By going through the NaN values, we must come up with a way to replace them with non NaN values or we need to come up with a reason for having NaN.

### **1. Android ver has 3 null values**

In [9]:
#The rows containing the null values in column Android Ver
ps_df[ps_df['Android Ver'].isnull()]

App         Category  Rating  \
4453                    [substratum] Vacuum: P  PERSONALIZATION     4.4   
4490                      Pi Dark [substratum]  PERSONALIZATION     4.5   
10472  Life Made WI-Fi Touchscreen Photo Frame              1.9    19.0   

      Reviews    Size Installs  Type     Price Content Rating  \
4453      230     11M   1,000+  Paid     $1.49       Everyone   
4490      189    2.1M  10,000+  Free         0       Everyone   
10472    3.0M  1,000+     Free     0  Everyone            NaN   

                  Genres    Last Updated Current Ver Android Ver  
4453     Personalization   July 20, 2018         4.4         NaN  
4490     Personalization  March 27, 2018         1.1         NaN  
10472  February 11, 2018          1.0.19  4.0 and up         NaN

In [10]:
#android ver has 3 null values
ps_df['Android Ver'].value_counts()

4.1 and up            2451
4.0.3 and up          1501
4.0 and up            1375
Varies with device    1362
4.4 and up             980
2.3 and up             652
5.0 and up             601
4.2 and up             394
2.3.3 and up           281
2.2 and up             244
4.3 and up             243
3.0 and up             241
2.1 and up             134
1.6 and up             116
6.0 and up              60
7.0 and up              42
3.2 and up              36
2.0 and up              32
5.1 and up              24
1.5 and up              20
4.4W and up             12
3.1 and up              10
2.0.1 and up             7
8.0 and up               6
7.1 and up               3
4.0.3 - 7.1.1            2
5.0 - 8.0                2
1.0 and up               2
7.0 - 7.1.1              1
4.1 - 7.1.1              1
5.0 - 6.0                1
2.2 - 7.1.1              1
5.0 - 7.1.1              1
Name: Android Ver, dtype: int64

Since we cannot replace the null values with any other values, lets drop the 3 rows which constitues only 0.03% of the total data of the column 

In [11]:
ps_df=ps_df[ps_df['Android Ver'].notna()]
ps_df.shape

(10838, 13)

We dropped the NaN values of Android Ver.

### **2. Current Ver has 8 Null values**

In [12]:
#The rows containing null values in the Current Ver column
ps_df[ps_df['Current Ver'].isnull()]

App             Category  Rating Reviews  \
15     Learn To Draw Kawaii Characters       ART_AND_DESIGN     3.2      55   
1553              Market Update Helper   LIBRARIES_AND_DEMO     4.1   20145   
6322            Virtual DJ Sound Mixer                TOOLS     4.2    4010   
6803                         BT Master               FAMILY     NaN       0   
7333                       Dots puzzle               FAMILY     4.0     179   
7407                   Calculate My IQ               FAMILY     NaN      44   
7730                            UFO-CQ                TOOLS     NaN       1   
10342                   La Fe de Jesus  BOOKS_AND_REFERENCE     NaN       8   

       Size    Installs  Type  Price Content Rating             Genres  \
15     2.7M      5,000+  Free      0       Everyone       Art & Design   
1553    11k  1,000,000+  Free      0       Everyone   Libraries & Demo   
6322   8.7M    500,000+  Free      0       Everyone              Tools   
6803   222k        100+  Free      0       Everyone          Education   
7333    14M     50,000+  Paid  $0.99       Everyone             Puzzle   
7407   7.2M     10,000+  Free      0       Everyone      Entertainment   
7730   237k         10+  Paid  $0.99       Everyone              Tools   
10342  658k      1,000+  Free      0       Everyone  Books & Reference   

            Last Updated Current Ver Android Ver  
15          June 6, 2018         NaN  4.2 and up  
1553   February 12, 2013         NaN  1.5 and up  
6322        May 10, 2017         NaN  4.0 and up  
6803    November 6, 2016         NaN  1.6 and up  
7333      April 18, 2018         NaN  4.0 and up  
7407       April 3, 2017         NaN  2.3 and up  
7730        July 4, 2016         NaN  2.0 and up  
10342   January 31, 2017         NaN  3.0 and up

In [13]:
#Finding the different values in the Current Ver column
ps_df['Current Ver'].value_counts()

Varies with device    1459
1.0                    809
1.1                    263
1.2                    178
2.0                    151
                      ... 
5.44.1                   1
7.16.8                   1
04.08.00                 1
2.10.06                  1
2.0.148.0                1
Name: Current Ver, Length: 2831, dtype: int64

We cannot replace the null values with any other values in the Current Ver column, So we have to drop the rows with null values

In [14]:
#Dropping the rows with null values in the Current Ver column
ps_df=ps_df[ps_df['Current Ver'].notna()]
ps_df.shape

(10830, 13)

### **3.Type column has only one null value**

In [15]:
#The row containiing the null value in the column Type
ps_df[ps_df['Type'].isnull()]

App Category  Rating Reviews                Size  \
9148  Command & Conquer: Rivals   FAMILY     NaN       0  Varies with device   

     Installs Type Price Content Rating    Genres   Last Updated  \
9148        0  NaN     0   Everyone 10+  Strategy  June 28, 2018   

             Current Ver         Android Ver  
9148  Varies with device  Varies with device

In [16]:
#Checking the values in the Type column
ps_df['Type'].value_counts()

Free    10032
Paid      797
Name: Type, dtype: int64

As the Type column has the type of the app i.e, paid or free and in the above row the price column has value 0, we can change the Type column null value to 'free'

In [17]:
#Changing the Type column null value to 'free'
ps_df.loc[9148,'Type']='Free'

In [18]:
#checking the result
ps_df[ps_df['Type'].isnull()]

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

We successfully replaced the null value of column Type to 'Free'

### **4.Rating column has 1474 null values**

In [19]:
#The rows containing the null values in the Rating column
ps_df[ps_df['Rating'].isnull()]


App             Category  Rating  \
23                  Mcqueen Coloring pages       ART_AND_DESIGN     NaN   
113              Wrinkles and rejuvenation               BEAUTY     NaN   
123                 Manicure - nail design               BEAUTY     NaN   
126           Skin Care and Natural Beauty               BEAUTY     NaN   
129    Secrets of beauty, youth and health               BEAUTY     NaN   
...                                    ...                  ...     ...   
10824                            Cardio-FR              MEDICAL     NaN   
10825                   Naruto & Boruto FR               SOCIAL     NaN   
10831             payermonstationnement.fr  MAPS_AND_NAVIGATION     NaN   
10835                             FR Forms             BUSINESS     NaN   
10838               Parkinson Exercices FR              MEDICAL     NaN   

      Reviews  Size  Installs  Type Price Content Rating  \
23         61  7.0M  100,000+  Free     0       Everyone   
113       182  5.7M  100,000+  Free     0   Everyone 10+   
123       119  3.7M   50,000+  Free     0       Everyone   
126       654  7.4M  100,000+  Free     0           Teen   
129        77  2.9M   10,000+  Free     0     Mature 17+   
...       ...   ...       ...   ...   ...            ...   
10824      67   82M   10,000+  Free     0       Everyone   
10825       7  7.7M      100+  Free     0           Teen   
10831      38  9.8M    5,000+  Free     0       Everyone   
10835       0  9.6M       10+  Free     0       Everyone   
10838       3  9.5M    1,000+  Free     0       Everyone   

                                Genres        Last Updated Current Ver  \
23     Art & Design;Action & Adventure       March 7, 2018       1.0.0   
113                             Beauty  September 20, 2017         8.0   
123                             Beauty       July 23, 2018         1.3   
126                             Beauty       July 17, 2018        1.15   
129                             Beauty      August 8, 2017         2.0   
...                                ...                 ...         ...   
10824                          Medical       July 31, 2018       2.2.2   
10825                           Social    February 2, 2018         1.0   
10831                Maps & Navigation       June 13, 2018   2.0.148.0   
10835                         Business  September 29, 2016       1.1.5   
10838                          Medical    January 20, 2017         1.0   

      Android Ver  
23     4.1 and up  
113    3.0 and up  
123    4.1 and up  
126    4.1 and up  
129    2.3 and up  
...           ...  
10824  4.4 and up  
10825  4.0 and up  
10831  4.0 and up  
10835  4.0 and up  
10838  2.2 and up  

[1470 rows x 13 columns]

Rating columns has 1474 null values so if we drop the rows we will loose a large amount of data.
In this case we should replace the null values with the median of the remaining values in the Rating column

In [20]:
#User defined function to find the median of the column Rating
def replace_with_median(series):
    return series.fillna(series.median())

ps_df['Rating'] = ps_df['Rating'].transform(replace_with_median)

In [22]:
#Now lets check the null values in the dataset Play Store Data
ps_df.isnull().sum()

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              0
Price             0
Content Rating    0
Genres            0
Last Updated      0
Current Ver       0
Android Ver       0
dtype: int64

# **Handling the duplicate values**

### **1. Handling the duplicate values in the App column**

In [21]:
ps_df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [24]:
ps_df['App'].value_counts()

ROBLOX                                                9
CBS Sports App - Scores, News, Stats & Watch Live     8
Candy Crush Saga                                      7
8 Ball Pool                                           7
ESPN                                                  7
                                                     ..
Meet U - Get Friends for Snapchat, Kik & Instagram    1
U-Report                                              1
U of I Community Credit Union                         1
Waiting For U Launcher Theme                          1
iHoroscope - 2018 Daily Horoscope & Astrology         1
Name: App, Length: 9649, dtype: int64

In [25]:
#Lets check the duplicate values 
ps_df[ps_df['App']=='ROBLOX']

App Category  Rating  Reviews Size      Installs  Type Price  \
1653  ROBLOX     GAME     4.5  4447388  67M  100,000,000+  Free     0   
1701  ROBLOX     GAME     4.5  4447346  67M  100,000,000+  Free     0   
1748  ROBLOX     GAME     4.5  4448791  67M  100,000,000+  Free     0   
1841  ROBLOX     GAME     4.5  4449882  67M  100,000,000+  Free     0   
1870  ROBLOX     GAME     4.5  4449910  67M  100,000,000+  Free     0   
2016  ROBLOX   FAMILY     4.5  4449910  67M  100,000,000+  Free     0   
2088  ROBLOX   FAMILY     4.5  4450855  67M  100,000,000+  Free     0   
2206  ROBLOX   FAMILY     4.5  4450890  67M  100,000,000+  Free     0   
4527  ROBLOX   FAMILY     4.5  4443407  67M  100,000,000+  Free     0   

     Content Rating                        Genres   Last Updated  \
1653   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
1701   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
1748   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
1841   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
1870   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
2016   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
2088   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
2206   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
4527   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   

       Current Ver Android Ver  
1653  2.347.225742  4.1 and up  
1701  2.347.225742  4.1 and up  
1748  2.347.225742  4.1 and up  
1841  2.347.225742  4.1 and up  
1870  2.347.225742  4.1 and up  
2016  2.347.225742  4.1 and up  
2088  2.347.225742  4.1 and up  
2206  2.347.225742  4.1 and up  
4527  2.347.225742  4.1 and up

In [33]:
ps_df[ps_df.duplicated()]

App      Category  Rating  \
229                       Quick PDF Scanner + OCR FREE      BUSINESS     4.2   
236                                                Box      BUSINESS     4.2   
239                                 Google My Business      BUSINESS     4.4   
256                                ZOOM Cloud Meetings      BUSINESS     4.4   
261                          join.me - Simple Meetings      BUSINESS     4.0   
...                                                ...           ...     ...   
8643                    Wunderlist: To-Do List & Tasks  PRODUCTIVITY     4.6   
8654   TickTick: To Do List with Reminder, Day Planner  PRODUCTIVITY     4.6   
8658                           ColorNote Notepad Notes  PRODUCTIVITY     4.6   
10049        Airway Ex - Intubate. Anesthetize. Train.       MEDICAL     4.3   
10768                                             AAFP       MEDICAL     3.8   

       Reviews                Size      Installs  Type Price Content Rating  \
229      80805  Varies with device    5,000,000+  Free     0       Everyone   
236     159872  Varies with device   10,000,000+  Free     0       Everyone   
239      70991  Varies with device    5,000,000+  Free     0       Everyone   
256      31614                 37M   10,000,000+  Free     0       Everyone   
261       6989  Varies with device    1,000,000+  Free     0       Everyone   
...        ...                 ...           ...   ...   ...            ...   
8643    404610  Varies with device   10,000,000+  Free     0       Everyone   
8654     25370  Varies with device    1,000,000+  Free     0       Everyone   
8658   2401017  Varies with device  100,000,000+  Free     0       Everyone   
10049      123                 86M       10,000+  Free     0       Everyone   
10768       63                 24M       10,000+  Free     0       Everyone   

             Genres       Last Updated         Current Ver         Android Ver  
229        Business  February 26, 2018  Varies with device        4.0.3 and up  
236        Business      July 31, 2018  Varies with device  Varies with device  
239        Business      July 24, 2018    2.19.0.204537701          4.4 and up  
256        Business      July 20, 2018      4.1.28165.0716          4.0 and up  
261        Business      July 16, 2018           4.3.0.508          4.4 and up  
...             ...                ...                 ...                 ...  
8643   Productivity      April 6, 2018  Varies with device  Varies with device  
8654   Productivity     August 6, 2018  Varies with device  Varies with device  
8658   Productivity      June 27, 2018  Varies with device  Varies with device  
10049       Medical       June 1, 2018              0.6.88          5.0 and up  
10768       Medical      June 22, 2018               2.3.1          5.0 and up  

[483 rows x 13 columns]

In [35]:
#Dropping duplicates from the column App
ps_df.drop_duplicates(subset='App',inplace=True)
ps_df.shape

(9649, 13)

In [36]:
#checking if the duplicate rows are dropped or not
ps_df[ps_df['App']=='ROBLOX']

App Category  Rating  Reviews Size      Installs  Type Price  \
1653  ROBLOX     GAME     4.5  4447388  67M  100,000,000+  Free     0   

     Content Rating                        Genres   Last Updated  \
1653   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   

       Current Ver Android Ver  
1653  2.347.225742  4.1 and up

We have dropped the duplicate values in the column App.Now the rows of the dataset are 9649.

### **Change the datatype of Last Updated column from string to datetime datatype**

In [42]:
ps_df['Last Updated']=pd.to_datetime(ps_df['Last Updated'])
ps_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9649 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   App             9649 non-null   object        
 1   Category        9649 non-null   object        
 2   Rating          9649 non-null   float64       
 3   Reviews         9649 non-null   object        
 4   Size            9649 non-null   object        
 5   Installs        9649 non-null   object        
 6   Type            9649 non-null   object        
 7   Price           9649 non-null   object        
 8   Content Rating  9649 non-null   object        
 9   Genres          9649 non-null   object        
 10  Last Updated    9649 non-null   datetime64[ns]
 11  Current Ver     9649 non-null   object        
 12  Android Ver     9649 non-null   object        
dtypes: datetime64[ns](1), float64(1), object(11)
memory usage: 1.0+ MB


### **Changing the datatype of Price column from string to float**

To change the datatype of Price column from string to float, first we have to remove the $ symbol from all the rows

In [43]:
ps_df['Price'].value_counts()

0          8896
$0.99       143
$2.99       124
$1.99        73
$4.99        70
           ... 
$18.99        1
$389.99       1
$19.90        1
$1.75         1
$1.04         1
Name: Price, Length: 92, dtype: int64

In [44]:
#Creating a userdefined function to drop the $ symbol
prices = [i for i in ps_df['Price']]

def clean_prices(prices_list):
    cleaned_data = []
    for price in prices_list:
        if '$' in price:
            price = price.replace('$', '')
        cleaned_data.append(price)
    return cleaned_data

ps_df['Price'] = clean_prices(prices)
ps_df['Price'] = ps_df['Price'].astype(float)

In [50]:
#Checking if the $ symbol is removed
ps_df[ps_df['Price']!=0].head()

App       Category  Rating  \
234  TurboScan: scan documents and receipts in PDF       BUSINESS     4.7   
235                 Tiny Scanner Pro: PDF Doc Scan       BUSINESS     4.8   
427                             Puffin Browser Pro  COMMUNICATION     4.0   
476                      Moco+ - Chat, Meet People         DATING     4.2   
477                                     Calculator         DATING     2.6   

    Reviews                Size  Installs  Type  Price Content Rating  \
234   11442                6.8M  100,000+  Paid   4.99       Everyone   
235   10295                 39M  100,000+  Paid   4.99       Everyone   
427   18247  Varies with device  100,000+  Paid   3.99       Everyone   
476    1545  Varies with device   10,000+  Paid   3.99     Mature 17+   
477      57                6.2M    1,000+  Paid   6.99       Everyone   

            Genres Last Updated  Current Ver Android Ver  
234       Business   2018-03-25        1.5.2  4.0 and up  
235       Business   2017-04-11        3.4.6  3.0 and up  
427  Communication   2018-07-05  7.5.3.20547  4.1 and up  
476         Dating   2018-06-19      2.6.139  4.1 and up  
477         Dating   2017-10-25        1.1.6  4.0 and up

In [49]:
#checking if the datatype of Price column is changed to float
ps_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9649 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   App             9649 non-null   object        
 1   Category        9649 non-null   object        
 2   Rating          9649 non-null   float64       
 3   Reviews         9649 non-null   object        
 4   Size            9649 non-null   object        
 5   Installs        9649 non-null   object        
 6   Type            9649 non-null   object        
 7   Price           9649 non-null   float64       
 8   Content Rating  9649 non-null   object        
 9   Genres          9649 non-null   object        
 10  Last Updated    9649 non-null   datetime64[ns]
 11  Current Ver     9649 non-null   object        
 12  Android Ver     9649 non-null   object        
dtypes: datetime64[ns](1), float64(2), object(10)
memory usage: 1.3+ MB


### **Changing the datatype of Install column from string to integer**

In [52]:
#Checking the contents of Install column
ps_df['Installs'].value_counts()

1,000,000+        1416
100,000+          1112
10,000+           1029
10,000,000+        937
1,000+             886
100+               709
5,000,000+         607
500,000+           504
50,000+            468
5,000+             467
10+                384
500+               328
50+                204
50,000,000+        202
100,000,000+       188
5+                  82
1+                  67
500,000,000+        24
1,000,000,000+      20
0+                  14
0                    1
Name: Installs, dtype: int64

To change the datatype of the Installs column, we first need to remove the + symbol from the Installs column

In [53]:
installs = [i for i in ps_df['Installs']]

def clean_installs(installs_list):
    cleaned_data = []
    for install in installs_list:
        if ',' in install:
            install = install.replace(',', '')
        if '+' in install:
            install = install.replace('+', '')
        install = int(install)
        cleaned_data.append(install)
    return cleaned_data
        
ps_df['Installs'] = clean_installs(installs)


In [54]:
#Checking if the '+' and ',' symbols from the Installs column
ps_df['Installs'].loc[10:20]

10     1000000
11     1000000
12    10000000
13      100000
14      100000
16      500000
17       10000
18     5000000
19    10000000
20      100000
Name: Installs, dtype: int64

We successfully converted the datatype of Installs column from string to Integer 